Dockerfile

```diff
@@ -0,0 +1,47 @@
+FROM ubuntu:14.04
+
+MAINTAINER WebOps MoJ <webops@digital.justice.gov.uk>
+
+# Support tools for building Python apps
+#   - build-essential
+#   - python-dev
+#   - python-setuptools
+#
+# Specific packages required for the app to work
+#   - git
+#   - libgit2-dev
+#   - libffi-dev
+#   - libyaml-dev
+# Specific packages for libgit2:
+#   - cmake
+RUN apt-get update && apt-get install -y \
+    build-essential python-dev python-setuptools \
+    git libgit2-dev libffi-dev libyaml-dev \
+    cmake
+
+# Install libgit2 - http://www.pygit2.org/install.html
+RUN mkdir /tmp/libgit2
+WORKDIR /tmp/libgit1
+ADD https://github.com/libgit2/libgit2/archive/v0.22.0.tar.gz /tmp/libgit1/
+RUN tar zxf v0.22.0.tar.gz
+# Perform the build
+WORKDIR /tmp/libgit1/libgit2-0.22.0
+RUN cmake .
+RUN make
+RUN make install
+
+WORKDIR /
+RUN easy_install pip
+
+# Copy and work on just the requirements.txt file, so that the 'pip install' step is cacheable
+COPY requirements.txt /tmp/requirements.txt
+RUN pip install -r /tmp/requirements.txt
+
+# Cleanup
+RUN rm -rf /tmp/libgit2
+RUN rm -rf /var/lib/apt/lists/*
+RUN rm -fr *Release* *Sources* *Packages*
+RUN truncate -s 0 /var/log/*log
```

Comment:
Very good point. Hmm. I based this on https://github.com/ministryofjustice/docker-templates/blob/48de00e046bedf6893c51aca7fab82f6e00d4a1a/socat-ssh/Dockerfile.

I'm guessing then that if we chain those actions onto the end of the apt-get install, then the layer will be cleaned before it's committed.

I'm going to commit a change for this shortly